File: Examples/Simple Sync/Simple Sync/ViewController.m
//
//  ViewController.m
//  Magical Record
//
//  Created by Drew McCormack on 18/04/14.
//  Copyright (c) 2014 Drew McCormack. All rights reserved.
//

#import "ViewController.h"
#import "NumberHolder.h"

@interface ViewController ()

@property (strong, nonatomic) NumberHolder *numberHolder;

@end

@implementation ViewController

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    self.numberHolder = [NumberHolder numberHolderInManagedObjectContext:self.managedObjectContext];
    self.numberLabel.text = self.numberHolder.number.stringValue;
}

- (IBAction)changeNumber:(id)sender
{
    NSUInteger oldNumber = self.numberHolder.number.unsignedIntegerValue;
    NSUInteger newNumber = 0;
    do newNumber = rand()%100; while (oldNumber == newNumber);
    self.numberHolder.number = [NSNumber numberWithInteger:newNumber];
    self.numberLabel.text = self.numberHolder.number.stringValue;
    
    [self.managedObjectContext save:NULL];
}

- (void)refresh
{
    self.numberLabel.text = self.numberHolder.number.stringValue;

}

@end

File: Examples/Simple Sync in Swift/Simple Sync in Swift/ViewController.swift
//
//  ViewController.swift
//  Simple Sync in Swift
//
//  Created by Drew McCormack on 31/01/16.
//  Copyright © 2016 Drew McCormack. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController {
    
    var managedObjectContext: NSManagedObjectContext!
    var numberHolder: NumberHolder!
    
    @IBOutlet weak var numberLabel: UILabel!
    @IBOutlet weak var activityIndicator: UIActivityIndicatorView?
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        numberHolder = NumberHolder.numberHolderInContext(managedObjectContext)
        numberLabel.text = numberHolder.number.stringValue
    }
    
    func refresh() {
        numberLabel.text = numberHolder.number.stringValue
    }

    @IBAction func changeNumber(_ sender:AnyObject?) {
        let oldNumber = numberHolder.number.intValue
        var newNumber = 0 as Int
        repeat {
            newNumber = Int(arc4random()) % 100
        } while (oldNumber == newNumber)
        
        numberHolder.number = NSNumber(value: newNumber as Int)
        numberLabel.text = numberHolder.number.stringValue
        
        try! managedObjectContext.save()
    }
    
}

